x-pack/plugins/discover_log_explorer/public/customizations/log_explorer_profile.tsx

```diff
@@ -19,11 +19,29 @@ interface CreateLogExplorerProfileCustomizationsDeps {
 export const createLogExplorerProfileCustomizations =
   ({ core }: CreateLogExplorerProfileCustomizationsDeps): CustomizationCallback =>
   async ({ customizations, stateContainer }) => {
-    const { DatasetsService } = await import('../services/datasets');
+    // Lazy load dependencies
+    const datasetServiceModuleLoadable = import('../services/datasets');
+    const logExplorerMachineModuleLoadable = import('../state_machines/log_explorer_profile');
+
+    const [{ DatasetsService }, { initializeLogExplorerProfileStateService, waitForState }] =
```

Comment:
**note:** parallelize the lazy loading for the required deps to keep the bundle size the smallest possible.